plugin/parser/engine/pg/deparse.go

```diff
@@ -14,22 +14,35 @@ import (
 func deparse(context parser.DeparseContext, in ast.Node, buf *strings.Builder) error {
 	switch node := in.(type) {
 	case ast.DataType:
-		return deparseDataType(context, node, buf)
+		if err := deparseDataType(context, node, buf); err != nil {
+			return err
+		}
 	case *ast.CreateTableStmt:
-		return deparseCreateTable(context, node, buf)
+		if err := deparseCreateTable(context, node, buf); err != nil {
+			return err
+		}
 	case *ast.AlterTableStmt:
-		return deparseAlterTable(context, node, buf)
+		if err := deparseAlterTable(context, node, buf); err != nil {
+			return err
+		}
 	case *ast.TableDef:
-		return deparseTableDef(context, node, buf)
+		if err := deparseTableDef(context, node, buf); err != nil {
+			return err
+		}
 	case *ast.ColumnDef:
-		return deparseColumnDef(context, node, buf)
+		if err := deparseColumnDef(context, node, buf); err != nil {
+			return err
+		}
 	case *ast.CreateSchemaStmt:
-		return deparseCreateSchema(context, node, buf)
+		if err := deparseCreateSchema(context, node, buf); err != nil {
+			return err
+		}
 	case *ast.DropSchemaStmt:
-		return deparseDropSchema(context, node, buf)
+		if err := deparseDropSchema(context, node, buf); err != nil {
+			return err
+		}
 	}
-
-	return errors.Errorf("failed to deparse %T", in)
+	return buf.WriteByte(';')
```

Comment:
We can't do this here. We will support deparse the part of the SQL statements, such as `deparseDataType`.
They don't need the `;`

Reply:
Done.